### PR TITLE
feat(save): make log file/folder permissions configurable

### LINKED ISSUE #729: Make save-folder log file permissions configurable (currently hardcoded 0600)
### Summary

The `save` middleware writes per-execution log files (`<ts>_<job>.stdout.log`, `.stderr.log`, `.json`) with a hardcoded mode of `0o600`, and creates `save-folder` with `0o750`. There is no way to override this. This makes the captured job logs readable **only by the uid the Ofelia daemon runs as** (root by default), which breaks any workflow where a non-root operator, a shared group, or another tool needs to read the logs on the host.

Would you accept a config option (and/or umask honoring) to make the mode configurable?

### Where

`middlewares/save.go` (current `main` and `v0.25.1`):

```go
os.MkdirAll(m.SaveFolder, 0o750)
os.WriteFile(filename, data, 0o600)
```

`SaveConfig` exposes `SaveFolder`, `SaveOnlyOnError`, `RestoreHistory`, `RestoreHistoryMaxAge` — no mode field. So there's no INI/label/env way to influence it.

### Background

These where changed in commit `5211e54b` ("security: address gosec findings (timeouts, perms, annotations)"):

```diff
- os.MkdirAll(m.SaveFolder, 0o755)     + os.MkdirAll(m.SaveFolder, 0o750)
- os.WriteFile(filename, data, 0o644)  + os.WriteFile(filename, data, 0o600)
```

i.e. just clearing gosec **G301/G306** (which cap file modes at `0600` by default). Before that, files were `0644` and the folder `0755`.

This is also a behavioral regression for anyone migrating from `mcuadros/ofelia`, which writes these files `0644` — after switching to this fork, the same `save-folder` logs become unreadable to non-root users with no documented cause.

### Why a config option (vs. just reverting)

`0600` is a reasonable *default* (job output can contain secrets/PII), so reverting outright would re-trip gosec and weaken the default. A configurable mode keeps the safe default while letting operators opt into broader readability for their environment.

### Proposed

A `save-mode` (file) and optionally `save-folder-mode` (directory) setting, parsed as octal, wired through `SaveConfig`:

```ini
[global]
save-folder = /var/log/ofelia/
save-mode = 0644          ; default 0600
save-folder-mode = 0755   ; default 0750
```

(Honoring the process umask instead of/along with an explicit mode would also work, but an explicit config key is more predictable across container runtimes.)

### Workaround today

A periodic root-side `chmod -R a+r <save-folder>`, since neither the job-container `user`/`default-user` nor a umask can affect these files — they're written by the daemon process, and an explicit `0600` create mode can't be widened by umask or POSIX default ACLs.

I'm happy to send a PR if the approach sounds good.

Environment: ofelia `v0.25.1`, `job-run` (ephemeral containers), `save-folder` via INI.

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Makes the `save` middleware's output permissions configurable, resolving #729.

Since commit `5211e54b` (clearing gosec G301/G306), the middleware hardcoded log files to `0600` and the save folder to `0750`, with no override. That makes captured job logs readable only by the daemon uid (root by default) — breaking any workflow where a non-root operator or shared group needs to read the logs on the host, and a behavioral regression vs. `mcuadros/ofelia` (which wrote `0644`).

Rather than revert (which would re-trip gosec and weaken the default for everyone), this adds opt-in config keys while keeping the hardened defaults.

## Changes

Two new optional `[global]`/per-job INI keys, parsed as octal:

```ini
save-folder = /var/log/ofelia/
save-mode = 0644          ; default 0600 (files: *.stdout.log/*.stderr.log/*.json)
save-folder-mode = 0755   ; default 0750 (the save folder)
```

- Accept `0644`, `0o644` or `644`; capped at `0777` (special setuid/sticky bits rejected).
- Validated at write time by the resolvers (`GetSaveFileMode`/`GetSaveFolderMode`), which fail the save with a clear, attributable error — no silent fallback to a wrong mode. (Config-load validation is wired in for completeness but, like the existing `save-folder`/`email-to` keys, does not fire for the squash-embedded `SaveConfig`; the runtime resolver is the authoritative guard.)
- Inherit global→job exactly like `save-folder`.

## Security

`save-mode`/`save-folder-mode` are deliberately **not** in the global Docker-label allow-list (mirroring `save-folder`), so a scheduled container cannot change the daemon-wide default. A regression test (`TestGlobalLabelAllowList_OmitsSavePermissionKeys`) pins this.

Note: per standard `mkdir -p` semantics, `save-folder-mode` only applies when Ofelia **creates** the folder; an existing (e.g. bind-mounted) folder keeps its current permissions. Documented in README and CONFIGURATION.md.

## Tests

- Octal parser edge cases (`0o`-prefix, bare, special-bit/non-octal rejection, `0777` boundary).
- Default vs. custom modes asserted on disk, umask-neutralized (in a `//go:build unix` file — `syscall.Umask` and Unix mode bits don't exist on the Windows release target).
- Config-validator octal cases; allow-list omission regression test.
- Global→job inheritance of both keys, in `mergeSaveDefaults` and `mergeSaveGlobals` (label-fills-empty and INI-wins).
- Both runtime mode-resolver error branches (`save-mode` and `save-folder-mode`) assert an attributable error.
- Minimum-count guards on the parser case tables (per review feedback).

## Verification

`go build ./...`, full `go test ./...` (linux), `GOOS=windows go vet ./...`, and `golangci-lint` all pass locally. (The local pre-push smoke hook hangs in the unrelated `core/adapters/docker` integration package on my WSL2 Docker box; this PR touches no files under `core/`.)

Closes #729

### CODE REVIEW

File: middlewares/save_test.go [hunk 1]
```diff
@@ -203,6 +203,79 @@ func TestSaveConfig_GetRestoreHistoryMaxAge_NegativeReturnsDefault(t *testing.T)
...
+		{in: "644", want: 0o644},
+		{in: " 0755 ", want: 0o755},
+		{in: "0", want: 0},
+		{in: "0777", want: 0o777},
+		{in: "1777", wantErr: true}, // special bits rejected
+		{in: "0888", wantErr: true}, // not octal
+		{in: "abc", wantErr: true},
+		{in: "", wantErr: true},
+	}
+	for _, tc := range cases {
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply middlewares/save_test.go:251]: Done in 15dcbd1 — added `require.GreaterOrEqual(t, len(cases), 11, ...)` before the loop so an accidentally-emptied table fails loudly instead of passing silently.

File: config/validator_boundary_test.go [hunk 2]
```diff
@@ -361,6 +361,42 @@ func TestValidator2ValidateIntField(t *testing.T) {
...
+		{"octal with leading zero", "0644", false},
+		{"octal 0o prefix", "0o755", false},
+		{"bare octal", "750", false},
+		{"max perm bits", "0777", false},
+		{"special bits rejected", "1777", true},
+		{"non-octal digit", "0888", true},
+		{"garbage", "rwx", true},
+	}
+
+	for _, tt := range tests {
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply config/validator_boundary_test.go:387]: Done in 15dcbd1 — added a `if len(tests) < 8 { t.Fatalf(...) }` guard before the loop (this file uses plain `testing`, not testify, so matched the repo's `t.Fatalf` min-count idiom).
